src/initialize.js

```diff
@@ -127,6 +131,32 @@ export async function auth(requireUser, hydrateUser) {
     hydrateAuthenticatedUser();
   }
 }
+/*
+ * Set or overrides configuration through an API.
+ * This method allows runtime configuration.
+ * Set a basic configuration when an error happen and allow initError and display the ErrorPage.
+ */
+
+export async function runtimeConfig() {
+  try {
+    const { MFE_CONFIG_API_URL, PUBLIC_PATH } = getConfig();
+
+    if (MFE_CONFIG_API_URL) {
+      const apiConfig = { headers: { accept: 'application/json' } };
+      const apiService = await configureCache();
+
+      const url = parseUrlQueryParams(MFE_CONFIG_API_URL, [
+        { name: 'mfe', value: (PUBLIC_PATH.slice(1, -1) || window.location.host.split('.')[0]) },
```

Comment:
Hmm... I don't know if we can assume that the MFE name exists in the URL. 😬  I didn't notice this earlier.  While we use the PUBLIC_PATH to know where the MFE is served from, I don't think we can otherwise make any assumptions about what the URL is or what information it might contain.  Today an MFE can be deployed to any sub-domain or path and it doesnt' matter, it's self-contained.

I'm thinking about the use cases for runtime config now.  If we want to use it to change styling/naming at runtime because we want to support multi-tenancy on a single MFE deployment, do we expect our edx-platform configuration to have different MFE config for each tenant?  Or do we want to use runtime config to simply replace build-time configuration?

I think I'd rather see us add an `APP_ID` environment variable to MFEs to specify the identifier of an MFE for use cases like this where a service needs to know which MFE is calling, but I don't think that helps the multi-tenancy use case.  

What use cases does eduNEXT need to support here / how do we expect to configure an MFE for multi-tenancy, which I assume is a concern?

Reply:
This use case is set or overrides the MFE configuration at runtime, not covering the multitenancy case.

This is a proposal [roadmap](https://docs.google.com/document/d/1-FHIQmyeQZu3311x8eYUNMru4JX7Yb3UlqjmJxvM8do/edit#heading=h.m3u89l2084ecl) you can comment and discuss it.

Reply:
I add de `APP_ID`

Reply:
For the record, I think `APP_ID` is a great idea. +1.